File: concurrency/primes/primes_single.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int is_prime(int n) {

  int top, i;

  if (n < 2)  return 0;
  if (n == 2) return 1;

  top = sqrt(n);

  for (i=2; i<top+1; ++i) 
    if ((n % i) == 0) 
      return 0;
  return 1;
}

int main(int argc, char *argv[])
{
  int top;
  int i;

  top = atoi(argv[1]);

  for (i=2; i<top+1; ++i) {
    if (is_prime(i)) {
      printf("%d\n", i);
    }
  }
  return 0;
}

File: concurrency/primes/primes_single.py
#! /usr/bin/env python
import sys
import os
import math

def is_prime(n):

    if n < 2:  return False
    if n == 2: return True
            
    top = int(math.sqrt(n))
    for i in range(2, top+1):
        if n % i == 0:
            return False
    return True
        
if __name__ == '__main__':

    args = sys.argv[1:]
    if not args:
        sys.exit(1)
    top = int(args[0])
    for n in range(top):
        if is_prime(n):
            print n


    
